### PR TITLE
Respect result from VerifyEventSignatures

### PR DESCRIPTION
**Pull request by richvdh**
Don't swallow the error when verification fails

### CODE REVIEW

**Comment by richvdh** [comment]
wtf does "file is not goimported" mean?

**Comment by erikjohnston** [comment]
Possibly that the imports are in the wrong order?

File: federationtypes.go [hunk 1]
```diff
@@ -134,8 +136,10 @@ func (r RespState) Check(ctx context.Context, keyRing JSONVerifier) error {
 	}
 
 	// Check if the events pass signature checks.
+	logger.Infof("Checking event signatures for %d events of room state", len(allEvents))
 	if err := VerifyEventSignatures(ctx, allEvents, keyRing); err != nil {
-		return nil
+		logger.Warnf("Event signature check failed: %s", err)
```
Comment (erikjohnston) [inline federationtypes.go]: TBH I'd expect the caller to handle/log errors

Comment (richvdh) [reply federationtypes.go]: fair. removed.

**Pull request merged**